src/test/java/io/vavr/beanvalidation2/valueextraction/SeqValueExtractorTest.java

```diff
@@ -0,0 +1,118 @@
+ *  \  \/  /  /\  \  \/  /  /
+ *   \____/__/  \__\____/__/
+ *
+ * Copyright 2014-2019 Vavr, http://vavr.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.vavr.beanvalidation2.valueextraction;
+
+import io.vavr.collection.List;
+import io.vavr.collection.Seq;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Path;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.constraints.NotBlank;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SeqValueExtractorTest {
+
+    private Validator validator;
+
+    @Before
+    public void setUp() {
+        this.validator = Validation.buildDefaultValidatorFactory().getValidator();
+    }
+
+    private <T> void validateAndAssertNoViolations(T target) {
+        Collection<ConstraintViolation<T>> violations = validator.validate(target);
+        assertThat(violations).isEmpty();
+    }
+
+    private <T> void validateAndAssertSingleViolation(
+            T target, int position, String type
+    ) {
+        Collection<ConstraintViolation<T>> violations = validator.validate(target);
+
+        assertThat(violations).isNotEmpty().hasSize(1);
+
+        ConstraintViolation<T> violation = violations.iterator().next();
+        assertThat(violation.getPropertyPath()).isNotEmpty().hasSize(2);
+
+        assertThat(violation.getConstraintDescriptor().getAnnotation()).isInstanceOf(NotBlank.class);
+
+        Iterator<Path.Node> iterator = violation.getPropertyPath().iterator();
+
+        assertThat(violation.getPropertyPath().toString())
+                .isEqualTo("list[2].<" + type + " element>");
+
+        Path.Node parent = iterator.next();
+        assertThat(parent.getName()).isEqualToIgnoringCase("list");
+
+        Path.Node child = iterator.next();
+        assertThat(child.getName()).isEqualToIgnoringCase("<" + type + " element>");
+        assertThat(child.getIndex()).isEqualTo(position);
+    }
+
+    @Test
+    public void defaultConstructionShouldValidate() {
+        validateAndAssertNoViolations(new TestBean());
+    }
+
+    @Test
+    public void havingEmptyValueShouldNotValidate() {
+        TestBean bean = new TestBean();
+        bean.add("");
+        validateAndAssertSingleViolation(bean, 2, "seq");
```

Comment:
Does it should be `"seq"` or different depending on exact type: `"list", "seq", "set", "stream", "queue"`. Is `Traversable` also supported?

Reply:
it is `"list"` for java in the hibernate validator implementation ( see https://github.com/hibernate/hibernate-validator/blob/master/engine/src/main/java/org/hibernate/validator/internal/engine/path/NodeImpl.java ) which is the interface for all list like types so I figured "`seq`" was fine

Reply:
`Traversable` is essentially and `Iterable` type and in scala 2.13 they actually renamed it and I proposed the same for `vavr` 1.0.0

So to stay with "<iterable element>" as catchall for all others is fine I think. ( see again https://github.com/hibernate/hibernate-validator/blob/master/engine/src/main/java/org/hibernate/validator/internal/engine/path/NodeImpl.java )

Reply:
On the other hand: hibernate-validator is an implementation of bean validation of which there could be multiple. `NodeImpl` is from hibernate-validator and we should probably not put too much weight on how they do things over there.

In the end there are 3 sensible feedbacks for generic vavr types (implementing `Value<T>`:
- an error occured at a certain index of the vavr type (Seqs
- an error occured at a certain key of the vavr type (Maps)
- an error occured SOMEWHERE in the generic type (Iterables, could be a Set or an Either,....)

Beanvalidations `ValueReceiver` distinguishes between 4 types:
- value
- iterableValue
- indexedValue
- keyedValue

Maybe we should do the same: so here is what i came up with
- indexedValue with feedback "`<index element>`" for Seq and below
- keyedValue with feedback "`<map key>`" or "`<map value>`" for Map and MultiMap (!!!)
- iterableValue with feedabck "`<iterable element>`" for all Traversables
- value with feedback "`<nested element>`" or "`<value element>`" for all other `Value<T>`

actually Multimap is a bit of a challenge here. since it combines keyed with indexed/iterable somewhat so I am open for suggestions (or introduce an own feedback type)

Reply:
shame that `Value` will be deprecated soon, in this context I really liked the separation as outlined above.

but we should bear in mind this issue: https://github.com/vavr-io/vavr/issues/2447

Reply:
I vote for: 

 * `<iterable element>` for Traversables
 * `<sequence element>` for Seq and below
 * `<map key>` and `<map value>` for Map and Multimap,
 * `<element>` for Values

Reply:
I almost agree on your proposal. 

Not 100% sure about sequence though. The API says `indexedValue` because it is the only thing that matters. A collection whose elements can be retrieved via index, and whose errors can be debugged by knowing the index.

So we dont care about it being a sequence or not, just "a type that you can query via index" as opposed to iterable or map.
This type being called `Seq` in vavr is merely a "coincidence", imho.

On the other hand, hibernate validator also calls it "list element".

Maybe "`<indexed element>`" or am I thinking too abstract here? 

What do you think?

Reply:
I think that the [Vavr's Seq](https://static.javadoc.io/io.vavr/vavr/0.10.0/io/vavr/collection/Seq.html) is named directly after [Scala's Seq](https://www.scala-lang.org/api/2.12.3/scala/collection/Seq.html). Both of them describe themselves as sequential, not only indexed.

Scala's: 
> sequences always have a defined order of elements

Vavr's:
> immutable sequential data structures

I opt for name `<sequence element>` as the type is called Seq, and describe it self as sequential. Indexed isn't mentioned anywhere. Using the same name as type and Javadoc used is more straight forward and easier to understand for users.

Reply:
You convinced me. Let's go for sequence element here